### PR TITLE
[wip] use github auth

### PR DESCRIPTION
**Pull request by jacobian**
Rather than using Django's built-in auth, admins should auth against github.
- [x] add github auth
- [x] use `GithubTeamOAuth2` instead of `GithubOrganizationOAuth2`
- [ ] create the signup flow:
  - [ ] start page for creating an account
  - [ ] post-auth complete page
  - [ ] document how admins should add more perms to created users
- [ ] add "login with github" to admin page
- [ ] remove password auth entirely
- [ ] figure out and document initial setup flow
- [ ] add whitelist?
- [ ] test with other users to make sure this is right

### CODE REVIEW

**Comment by NoahKunin** [comment]
Why is MyUSA being considered? It's on the line to deprecation due to Login. I'd use GitHub auth until Login is up.

**Comment by jacobian** [comment]
github it is 👍

**Comment by jessieay** [comment]
@jacobian thoughts on scoping to https://github.com/orgs/18F/teams/18f so only 18F staff can auth rather than anyone in the 18F org? (197 vs 346 ppl)

**Comment by jacobian** [comment]
@jessieay it's hard to tell, but that's what this does, see https://github.com/18F/bug-bounty/pull/6/files#diff-041953c637f95e25fb35d51022891763R79 - using the `GithubOrganizationOAuth2` backend takes care of that. See http://psa.matiasaguirre.net/docs/backends/github.html#github-for-organizations for the (scant) documentation.

**Comment by jessieay** [comment]
@jacobian I was thinking you'd want `GithubTeamOAuth2.` -- I think you are using the 18F Org, which is bigger than the 18F Team

**Comment by jacobian** [comment]
@jessieay oh shit, I'd missed that difference! Yes, you're totally right - thank you!

**Comment by konklone** [comment]
@jacobian So how much do we pay @jessieay for that one? :smile_cat:

**Comment by NoahKunin** [comment]
@jessieay http://i.giphy.com/13B1WmJg7HwjGU.gif

File: bugbounty/settings.py [hunk 1]
```diff
@@ -70,3 +73,21 @@
...
+
+# python-social-auth config - use github
+AUTHENTICATION_BACKENDS = (
+    'social.backends.github.GithubTeamOAuth2',
+    'django.contrib.auth.backends.ModelBackend',
+)
+SOCIAL_AUTH_GITHUB_TEAM_KEY = os.environ.get('GITHUB_KEY', None)
+SOCIAL_AUTH_GITHUB_TEAM_SECRET = os.environ.get('GITHUB_SECRET', None)
+
+# The team ID doesn't seem to be easy to find on the web anywhere, so I
```
Comment (jessieay) [inline bugbounty/settings.py:85]: agreed that it's hard to find. I added some docs in dolores on how I did it: https://github.com/18F/dolores-landingham-bot/compare/jy-github-auth#diff-6a3371457528722a734f3c51d9238c13R127

Not sure if that seems easier or harder than what you're recommending

**Pull request closed without merging**